BraveShared/Analytics/UserReferralProgram.swift

```diff
@@ -23,6 +23,13 @@ public class UserReferralProgram {
         static let prod = "https://laptop-updates.brave.com"
     }
     
+    // In case of network problems when looking for referrral code
+    // we retry the call few times while the app is still alive.
+    private var referralLookupRetryTimer: Timer?
```

Comment:
Probably more clear to just wrap these all in a struct, since all related to the same thing.

```
struct ReferralLookupRetry {
    private var retryTimer: Timer?
    private var currentCount = 0
    ...
}
```